### PR TITLE
Web Inspector: Update jsmin to 3.0.1

### PR DESCRIPTION
**Pull request by philn**
#### 057f942c764e9f79d38e88bca94b7e35549883b4
<pre>
Web Inspector: Update jsmin to 3.0.1
<a href="https://bugs.webkit.org/show_bug.cgi?id=239924">https://bugs.webkit.org/show_bug.cgi?id=239924</a>

Patch by Philippe Normand &lt;pnormand@igalia.com &gt; on 2022-05-04
Reviewed by Yusuke Suzuki.

Updated jsmin from upstream version, applying a few style changes suggested by
check-webkit-style. This new version is Python3-only and also much faster. Old:

time python /app/webkit/Source/JavaScriptCore/Scripts/jsmin.py &lt; /app/webkit/WebKitBuild/Release/WebInspectorUI/DerivedSources/Main.js  &gt; foo.js

real 1m21.234s
user 0m59.580s
sys  0m21.253s

New:

time python /app/webkit/Source/JavaScriptCore/Scripts/jsmin.py &lt; /app/webkit/WebKitBuild/Release/WebInspectorUI/DerivedSources/Main.js  &gt; foo.js

real 0m3.933s
user 0m3.899s
sys  0m0.018s

* Scripts/jsmin.py:
(jsmin):
(JavascriptMinify.__init__):
(JavascriptMinify.minify.write):
(JavascriptMinify.minify):
(JavascriptMinify):
(JavascriptMinify.regex_literal):
(JavascriptMinify.regex_literal.cannot):
(JavascriptMinify.line_comment):
(JavascriptMinify.block_comment):
(JavascriptMinify.newline):
(JavascriptMinify.minify.read): Deleted.

LayoutTests:
* platform/gtk/inspector/timeline/line-column-expected.txt:

Canonical link: <a href="https://commits.webkit.org/250246@main">https://commits.webkit.org/250246@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@293767">https://svn.webkit.org/repository/webkit/trunk@293767</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Source/JavaScriptCore/Scripts/jsmin.py [hunk 1]
```diff
@@ -199,46 +126,130 @@ def read(n):
...
+                    if next2 in self.newlinestart_strings \
+                            or next2 > '~' or next2 == '/':
+                        do_newline = True
+                    break
+
+        return next2, do_newline
+
+
 if __name__ == '__main__':
+    import sys
```
Comment (JonWBedard) [inline Source/JavaScriptCore/Scripts/jsmin.py]: Why isn't this at the top-level?

File: Source/JavaScriptCore/ChangeLog [hunk 2]
```diff
@@ -1,3 +1,40 @@
...
+
+        time python /app/webkit/Source/JavaScriptCore/Scripts/jsmin.py < /app/webkit/WebKitBuild/Release/WebInspectorUI/DerivedSources/Main.js > foo.js
+
+        real 1m21.234s
+        user 0m59.580s
+        sys  0m21.253s
+
+        New:
+
+        time python /app/webkit/Source/JavaScriptCore/Scripts/jsmin.py < /app/webkit/WebKitBuild/Release/WebInspectorUI/DerivedSources/Main.js > foo.js
```
Comment (JonWBedard) [inline Source/JavaScriptCore/ChangeLog]: Presumably `python` on your system is `python3`, correct?

File: Source/JavaScriptCore/Scripts/jsmin.py [hunk 3]
```diff
@@ -24,40 +26,20 @@
 # THE SOFTWARE.
 
 
-import sys
-is_3 = sys.version_info >= (3, 0)
```
Comment (JonWBedard) [inline Source/JavaScriptCore/Scripts/jsmin.py]: I think stripping Python 2 support is the right call, but that's something more JavaScriptCore folks should verify

File: Source/JavaScriptCore/Scripts/jsmin.py [hunk 4]
```diff
@@ -1,3 +1,5 @@
+# vim: set fileencoding=utf-8 :
+
 # This code is original from jsmin by Douglas Crockford, it was translated to
```
Comment (JonWBedard) [inline Source/JavaScriptCore/Scripts/jsmin.py:3]: No license update?

File: Source/JavaScriptCore/ChangeLog (hunk 2 above)
Comment (philn) [reply Source/JavaScriptCore/ChangeLog]: Yes, the Flatpak SDK no longer ships Python 2 :)

File: Source/JavaScriptCore/Scripts/jsmin.py (hunk 4 above)
Comment (philn) [reply Source/JavaScriptCore/Scripts/jsmin.py:3]: No, there were just a couple white space changes, that I removed to reduce noise in the diff.

File: Source/JavaScriptCore/Scripts/jsmin.py (hunk 1 above)
Comment (philn) [reply Source/JavaScriptCore/Scripts/jsmin.py]: For no particular reason

File: Source/JavaScriptCore/ChangeLog (hunk 2 above)
Comment (JonWBedard) [reply Source/JavaScriptCore/ChangeLog]: For the sake of Apple folks, can we make this `python3`?

Comment (philn) [reply Source/JavaScriptCore/ChangeLog]: In the script shebang you mean? Actually currently there's none at all, I assumed python3 was the default one now, but I can make it explicit in the shebang.

**Review by Constellation — APPROVED** [review]
Can you check the generated output is the same? If it is, then r=me.

File: Source/JavaScriptCore/ChangeLog (hunk 2 above)
Comment (JonWBedard) [reply Source/JavaScriptCore/ChangeLog]: I mean in your commit message with the timing.

**Comment by philn** [comment]
> Can you check the generated output is the same? If it is, then r=me.

The new minified files are a bit smaller now actually, less crlf and space characters. Diff attached.
[mini.diff.txt](https://github.com/WebKit/WebKit/files/8612368/mini.diff.txt)

**Comment by Constellation** [comment]
Looks good.

**Pull request merged**